site/source/docs/contributing/developers_guide.rst

```diff
@@ -36,37 +57,46 @@ Patches should be submitted as *pull requests* to the **incoming** branch.
-- :ref:`Run all the automatic tests <emscripten-test-suite>` and make sure they pass.  Some tests might not be required for very simple patches (for example, when just adding tests for new library functions). If you don't have time to run all the tests, at least run a random subset of them (say, 100), see that link.
-- "Add an automatic test to `tests/runner.py <https://github.com/emscripten-core/emscripten/blob/master/tests/runner.py>`_ if you add any new functionality or fix a bug.
-- Record the tests that were run in the pull request or issue.
-
+- Add an automatic test if you add any new functionality or fix a bug. Search
+  in ``tests/*.py`` for related tests, as often the simplest thing is to add to
+  an existing one. If you're not sure how to test your code, feel free to ask
+  for help.
+- We normally squash and merge PRs, which means the PR turns into a single
+  commit on the target branch. Because of that, it's ok to have merge commits
+  in the PR itself, as they get removed. Please put a good description for
+  the final commit in the PR description, and we'll use it when squashing.
 
 Code reviews
 ============
 
-`@kripken <https://github.com/kripken>`_ reviews all pull requests before merging.
-
-Exceptions are sub-projects that are 'owned' by other people. These owners can push to incoming directly:
-
--  OpenAL and audio in general: `@ehsan <https://github.com/ehsan>`_
--  embind: `@imvu <https://github.com/imvu>`_
--  Windows stuff: `@juj <https://github.com/juj>`_
-
+One of the core developers will review a pull request before merging it. If
+several days pass without any comments on your PR, please comment in the PR
+which will ping them. (If that happens, sorry! Sometimes things get missed.)
 
 Compiler overview
 =================
 
 The :ref:`Emscripten Compiler Frontend (emcc) <emccdoc>` is a python script that manages the entire compilation process:
 
-- *emcc* calls :term:`Clang` to convert C++ to bitcode, ``llvm-opt`` to optimize it, ``llvm-link`` to link it, etc.
-- *emcc* then calls `emscripten.py <https://github.com/emscripten-core/emscripten/blob/master/emscripten.py>`_, which performs any final compilation to wasm (including invoking *wasm-emscripten-finalize* from Binaryen, for wasm) and emits the JavaScript.
-- If optimizing wasm, *emcc* will then call Binaryen's optimizer, run meta-dce, and other useful things. It will also run JS optimizations on the JS that is emitted alongside the wasm.
+- **emcc** calls :term:`Clang` to compile C++ and ``wasm-ld`` to link it. It
+  builds and integrates with the Emscripten system libraries, both the
+  compiled ones and the ones implemented in JS.
+- **emcc** then calls `emscripten.py <https://github.com/emscripten-core/emscripten/blob/master/emscripten.py>`_
+  which performs the final transformation to wasm (including invoking
+  **wasm-emscripten-finalize** from Binaryen) and calls the JS compiler
```

Comment:
Is wasm-emscripten-finalize described somewhere? If so, it would be useful to link to it, if not, then a short high-level description might be helpful too.

Reply:
It's an internal tool really, I don't think we have any more docs than this page. Maybe the tools `--help` output has a little more info...

I can add some description here, but not sure what kind of text you're thinking of?

Reply:
I could try to make a suggestion, but tbh I don't know what it does myself, hence the suggestion to add some description for all potential contributors :) "Extracts metadata and generates JavaScript bindings"? (again, I have no idea if this is what it actually does)